### ISSUE TITLE
[BUG] semi-foundation contain react framework type

### ISSUE THREAD

**Issue by rashagu** [DouyinFE/semi-design#923]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Which Component

datePicker ，may be many more components

### Semi Version

2.13.0

### Current Behavior

```
../../node_modules/.pnpm/@douyinfe+semi-foundation@2.13.0/node_modules/@douyinfe/semi-foundation/datePicker/monthsGridFoundation.ts:43:15 - error TS2503: Cannot find namespace 'React'.

43     navPrev?: React.ReactNode;
                 ~~~~~

../../node_modules/.pnpm/@douyinfe+semi-foundation@2.13.0/node_modules/@douyinfe/semi-foundation/datePicker/monthsGridFoundation.ts:44:15 - error TS2503:
 Cannot find namespace 'React'.

44     navNext?: React.ReactNode;
                 ~~~~~

../../node_modules/.pnpm/@douyinfe+semi-foundation@2.13.0/node_modules/@douyinfe/semi-foundation/datePicker/monthsGridFoundation.ts:45:24 - error TS2503:
 Cannot find namespace 'React'.

45     renderDate?: () => React.ReactNode;
                          ~~~~~

../../node_modules/.pnpm/@douyinfe+semi-foundation@2.13.0/node_modules/@douyinfe/semi-foundation/datePicker/monthsGridFoundation.ts:46:28 - error TS2503:
 Cannot find namespace 'React'.

46     renderFullDate?: () => React.ReactNode;
                              ~~~~~

../../node_modules/.pnpm/@douyinfe+semi-foundation@2.13.0/node_modules/@douyinfe/semi-foundation/datePicker/foundation.ts:128:21 - error TS2503: Cannot find namespace 'React'.

128     dropdownStyle?: React.CSSProperties;
                        ~~~~~

../../node_modules/.pnpm/@douyinfe+semi-foundation@2.13.0/node_modules/@douyinfe/semi-foundation/datePicker/foundation.ts:133:18 - error TS2503: Cannot find namespace 'React'.

133     inputStyle?: React.CSSProperties;
                     ~~~~~

../../node_modules/.pnpm/@douyinfe+semi-foundation@2.13.0/node_modules/@douyinfe/semi-foundation/datePicker/foundation.ts:149:13 - error TS2503: Cannot find namespace 'React'.

149     style?: React.CSSProperties;
                ~~~~~

../../node_modules/.pnpm/@douyinfe+semi-foundation@2.13.0/node_modules/@douyinfe/semi-foundation/datePicker/inputFoundation.ts:48:18 - error TS2503: Cannot find namespace 'React'.

48     inputStyle?: React.CSSProperties;

```

### Expected Behavior

Check by ts

### Environment

```markdown
- OS: win11
- browser: edge 100
```

**Comment by pointhalo** [comment]
@shijiatongxue

**Comment by pointhalo** [comment]
fix 2.14.0-beta.0
